external-crates/move/crates/move-package/src/lib.rs

```diff
@@ -102,6 +102,10 @@ pub struct BuildConfig {
     /// If `true`, disable linters
     #[clap(long, global = true)]
     pub no_lint: bool,
+
+    #[clap(skip)]
+    /// A field specifying this compiler version, derived from sui-move crate version.
+    pub compiler_version: Option<String>,
```

Comment:
We need to pass the compiler version to crates in `external-crates`. Changing the `Cargo.toml` in `external-crates` to use `version.workspace = true` does not work (it only works in the `sui/crates` directory. I made the choice to persist this in `BuildConfig` which I figured is most suitable. It's kinda ugly, but passing it as an argument through deep layers of functions is very invasive. Maybe there's a way to have `version.workspace = true` work for `external-crates` but I couldn't figure it out.